api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto

```diff
@@ -206,6 +205,35 @@ message ExternalProcessor {
   // Instead, the stream to the external processor will be closed. There will be no
   // more external processing for this stream from now on.
   bool disable_immediate_response = 15;
+
+  // Options related to the sending and receiving of dynamic metadata
+  MetadataOptions metadata_options = 16;
+}
+
+// The MetadataOptions structure defines options for the sending and receiving of
+// dynamic metadata. Specifically, which namespaces to send to the server, whether
+// metadata returned by the server may be written, and how that metadata may be written.
+message MetadataOptions {
+  message MetadataNamespaces {
+    // Specifies a list of metadata namespaces whose values, if present,
+    // will be passed to the ext_proc service as an opaque *protobuf::Struct*.
+    repeated string untyped = 1;
+
+    // Specifies a list of metadata namespaces whose values, if present,
+    // will be passed to the ext_proc service as a *protobuf::Any*. This allows
+    // envoy and the external processing server to share the protobuf message
+    // definition for safe parsing.
+    repeated string typed = 2;
+  }
+
+  // Describes which typed or untyped dynamic metadata namespaces to forward to
```

Comment:
I'm not very familiar with this filter, what exactly does "metadata namespaces" mean here and what is done with them?

Reply:
The [`config.core.v3.Metadata`](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/base.proto.html) contains untyped and typed metadata implemented as `map<string,Struct>` and `map<string,Any>` respectively. The namespace is the typical term used to refer to the key in this map under which the metadata is stored. If these namespaces are configured here, the contents of the map at that key will be included in the request to the external processor.

Reply:
Got it, thanks for the explanation!